### PR TITLE
Add debugger RFC

### PR DESCRIPTION
**Pull request by logicminds**
Adds an RFC for a new pdk debugger sub command.

### CODE REVIEW

**Comment by scotje** [comment]
@logicminds thank you for your proposal, it looks really solid!

I'll be adding a few comments on things as I have time, and I'll try to get some other 👀 on it as well.

I'd love to discuss this more extensively at an upcoming PDK Community Planning meeting on the Puppet Community Slack. The regular schedule for those is every other Tuesday from 5-6 PM Pacific time. (That time range works best for our team members in Australia.) The next meeting is Tuesday, August 27. Would you be available to discuss this during that meeting or perhaps the following meeting on Sept 10?

If that time of day is not compatible with your schedule, we may be able to schedule an ad-hoc planning session at an alternate time as well, or just discuss this asynchronously on the PR.

**Comment by logicminds** [comment]
Asynchronous communications work best as I travel a bunch.  I'll be on the east coast during the the next PDK meeting which I should be able to attend as well.

File: RFCs/0000-rfc-add-pdk-dubugger.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+Add the [puppet-debug](https://forge.puppet.com/nwops/debug) module to the fixtures by default. See the [following article](http://logicminds.github.io/blog/2017-04-25-break-into-your-puppet-code/) for some use cases about inserting a break statement  (similar to binding.pry).
+
+
+## Drawbacks
+The debugger is a community developed gem and will have updates every so often. These updates may break the `pdk console` command or include dependencies that conflict in some way. Testing will need to be added to the PDK to account for future updates.  However, this is no different than any other gem included with the pdk. 
+
+## Alternatives
+
+The alternative is to list the debugger as a dependency in the `puppet-module-posix-` gems and the user can invoke
```
Comment (glennsarti) [inline RFCs/0000-rfc-add-pdk-dubugger.md]: nit: This should not stop merging, but:

> `puppet-module-posix-`

should be

> `puppet-module-posix-` and `puppet-module-windows-`

File: RFCs/0000-rfc-add-pdk-dubugger.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+There are two config options that are specific to PDK that make the debugger's involvement special.  Without the PDK the debugger cannot switch ruby or puppet versions easily.  The PDK makes this simple using the 
+`--puppet-version=5.5.12` or `--pe-version=2018.1` arguments.  These arguments will allow the user to use the debugger to see how the language changed between versions. 
+
+`pdk console --puppet-version=5.5.12 -e 'true' =~ Boolean'`
+
+`pdk console --puppet-version=6.7.0 -e "'true' =~ Boolean"`
+
+This would tell the user if the code has changed behavior in the newer version.  
+
+### Add to puppet-module-posix- gems
```
Comment (glennsarti) [inline RFCs/0000-rfc-add-pdk-dubugger.md]: Hrmm, this would stop merging... what about Windows?

Comment (logicminds) [reply RFCs/0000-rfc-add-pdk-dubugger.md]: Is there a correct name for these gems that addresses both platforms?

Comment (glennsarti) [reply RFCs/0000-rfc-add-pdk-dubugger.md]: `Puppet Module Meta-gems`

`puppet-module meta gems`  <--- My pick

`puppet-module-* gems`

Comment (glennsarti) [reply RFCs/0000-rfc-add-pdk-dubugger.md]: ping @rodjek  @bmjen  👆👆

Comment (logicminds) [reply RFCs/0000-rfc-add-pdk-dubugger.md]: Fixed with the updated gem name.

**Comment by scotje** [comment]
Entered Final Comment Period with an intent to accept if there are no major objections.

**Pull request merged**